### PR TITLE
fix:remove RevalCurrentPmSyncJob from nimdta

### PR DESCRIPTION
**Pull request by hihilary**
Use `@Profile("!nimdta")` instead of "@ConditionXXX" to make everywhere consistent. I believe this would be easier for us to locate them in the future. But it requires the profile to contain `nimdta` first. 

depends on [TIS-OPS#507](https://github.com/Health-Education-England/TIS-OPS/pull/507)

TIS21-3229

### CODE REVIEW

File: src/main/java/uk/nhs/tis/sync/api/JobResource.java [hunk 1]
```diff
@@ -84,10 +92,17 @@ public ResponseEntity<Map<String, Boolean>> getStatus() {
...
-    statusMap.put("revalCurrentPmJob", revalCurrentPmSyncJob.isCurrentlyRunning());
+    if (revalCurrentPmSyncJob != null) {
+      statusMap.put("revalCurrentPmJob", revalCurrentPmSyncJob.isCurrentlyRunning());
+    }
     return ResponseEntity.ok().body(statusMap);
   }
 
+  @GetMapping("/sys/profile")
+  public ResponseEntity<String> getSysProfile() {
+    return ResponseEntity.ok(this.activeProfile);
```
Comment (JosephKelly) [inline src/main/java/uk/nhs/tis/sync/api/JobResource.java:103]: JS syntax `this.` :hand_over_mouth:

File: src/main/java/uk/nhs/tis/sync/event/listener/JobRunningListener.java [hunk 2]
```diff
@@ -48,13 +48,17 @@ public class JobRunningListener implements ApplicationListener<ApplicationReadyE
...
 
   private LocalTime earliest;
 
   private LocalTime latest;
 
+  @Autowired(required = false)
+  public void setRevalCurrentPmSyncJob(RevalCurrentPmSyncJob revalCurrentPmSyncJob) {
+    this.revalCurrentPmSyncJob = revalCurrentPmSyncJob;
+  }
+
```
Comment (JosephKelly) [inline src/main/java/uk/nhs/tis/sync/event/listener/JobRunningListener.java:61]: All the other jobs use annotation on the fields.  Perhaps in a later PR, would you think about whether it makes more sense to use field injection or constructor+setter injection?

**Review by JosephKelly — CHANGES_REQUESTED** [review]
Minor tweaks that don't affect functionality... If you'd prefer me to pick them up next week, I'll happily do so!

**Comment by hihilary** [comment]
> Minor tweaks that don't affect functionality... If you'd prefer me to pick them up next week, I'll happily do so!

Yes, please take it @JosephKelly and thanks a lot!

**Review by JosephKelly — APPROVED** [review]
Soz... too many things going on.  Thought I had approved!

**Pull request merged**